### ISSUE TITLE
RabbitMQ MaaS checks failing Unauthorized

### ISSUE THREAD

**Issue by xeregin** [rcbops/rpc-openstack#1147]
##### What version (commit) do you encounter this issue on?
12.2rc3

##### What type of issue is this (bug, feature, documentation, regression)?
Bug

##### What is the expected behavior?
The Python plugin should authenticate to the API and make a request.

##### What is actually happening?
The Python plugin is failing to authenticate to the API, receiving a 401 error.

##### What were you attempting to do when you encountered this error (give as much detail as possible)?
Configured and deployed MaaS. The RabbitMQ checks all failed because the password in user_extras_secrets did not match the actual RabbitMQ password. This happened on both labs that we upgraded.

##### How can we reproduce this issue?
I believe that the RabbitMQ checks were working properly preupgrade. The labs were on 12.2rc2 and the issue came about after the upgrade to rc3.

##### How can we verify your issue is addressed?
By verifying that the 24 RabbitMQ alarms are passing (8 per infra node)

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?
This is not a large issue and definitely not a blocker. A workaround is to jump into the rabbit container on an infra node and type the following command:
```
rabbitmqctl change_password maas_user <password from secrets file>
```

##### What is your evaluation on the importance of this issue, and why?
Not a high priority, because there is a very basic workaround.

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.2

##### Does your change require documentation?
Unsure, but I believe:
- [x] No documentation required for this change

Tasks:

- [ ] Reproduce the issue. 
- [x] Create a PR for fix and/or docs.
- [ ] Incorporate feedback from pull request reviews.
- [ ]  What is the gating for this on DEV and/or QE? Add related gating if appropriate. Also make sure current gating passes with this fix.
- [ ] Reach out to reporter to determine if the fix is resolving the issue.
- [ ] What are the upgrade impacts (if any).
- [ ] What are the backports are needed (if any).

**Comment by hughsaunders** [comment]
Did the password in the u_e_secrets file change during the upgrade? or was it removed?

**Comment by git-harry** [comment]
As Hugh suggested, it looks like you removed the secrets file or 'maas_rabbitmq_password' from the file. To address this we would need to add 'force="yes"' [1] to the user setup task [2].

[1] http://docs.ansible.com/ansible/rabbitmq_user_module.html
[2] https://github.com/rcbops/rpc-openstack/blob/master/rpcd/playbooks/roles/rpc_maas/tasks/rabbitmq_user.yml

**Comment by xeregin** [comment]
I don't believe the password was changed during the upgrade, the u_e_secrets file is still present, and the maas_rabbitmq_password variable is still defined in both labs. However, @Rydor mentioned that Tempest may be changing the password, and for whatever reason neglecting to revert back.

**Comment by cloader89** [comment]
@Rydor is correct, I just checked the failure, and it's a tempest auth test causing the fail, I'll do some more looking into this.

**Comment by sigmavirus24** [comment]
Who reproduced this issue? Has anyone discovered why on an upgrade the RabbitMQ password changes if the secrets file doesn't? That seems to be the *real* bug here.

**Comment by mancdaz** [comment]
and I'm pretty sure that tempest has no knowledge of the rabbitmq maas user

**Comment by sigmavirus24** [comment]
> and I'm pretty sure that tempest has no knowledge of the rabbitmq maas user

Yeah, I don't know why @Rydor thinks Tempest is responsible for this. Perhaps they could explain their reasoning themselves?

**Comment by alextricity25** [comment]
@cloader89 and I talked about this. We weren't able to find what changed the rabbitmq maas user password, but we never tested with upgrades. I still thought adding "force: True" to the rabbitmq_user task wasn't such a bad idea. This would cover the case of the user accidentally or purposely changing the password.
@cloader89, I might have mislead you. I agree with @sigmavirus24 in that it would be good to do test an upgrade, then see what might have changed the rabbitmq maas user password, if anything. There might be something else in the works that we might have missed.

**Comment by sigmavirus24** [comment]
I thought "Reproduce issue" was completed when I looked earlier... Am I imagining things?

**Comment by cloader89** [comment]
@sigmavirus24  it was clicked mistakenly... So we tried reproducing the error, it was not reproduce-able. We went from r12.2.0rc2 to rc3. In between we copied the rc2 user_extras_secret file over to home directory to check if the files were modified or no during the rc3 upgrade.  We then used the rabbitmqctl command inside the container to authenticate using the maas_user. It successfully authenticated. To further check, we even ran tempest and it still connected. Therefore, it is not reproduce-able.
